extending-cli.md

```diff
@@ -0,0 +1,76 @@
+Extending the CLI
+================= 
+
+NativeScript CLI allows you to extend its behavior and customize it for your project. 
+When the CLI executes a particular command (for example `tns build`) it checks whether you have added such extending hooks and executes them.
+Additionally, plugins can use these hooks to control the compilation of the program.
+
+What are the hooks
+==================
+
+The hooks are any executable code or node.js scripts placed in the `hooks` subdirectory of your project.
+The hook file name must follow a strict naming scheme. The name declares:
+
+ - the action being hooked
+ - whether the user code must be called before or after the action
```

Comment:
Is this true? The TypeScript plugin puts subdirectories named `before_prepare` and `before_watch`, each containing a JavaScript file...